src/mito2/src/region.rs

```diff
@@ -371,14 +371,30 @@ impl ManifestContext {
         // Checks state inside the lock. This is to ensure that we won't update the manifest
         // after `set_readonly_gracefully()` is called.
         let current_state = self.state.load();
-        ensure!(
-            current_state == RegionRoleState::Leader(expect_state),
-            RegionLeaderStateSnafu {
-                region_id: manifest.metadata.region_id,
-                state: current_state,
-                expect: expect_state,
-            }
-        );
+
+        // If expect_state is writable, the current state must be either writable or downgrading.
+        //
+        // A downgrading leader rejects user writes but still allows
+        // flushing the memtable and updating the manifest.
+        if expect_state == RegionLeaderState::Writable {
```

Comment:
Will that be other states?

Reply:
resolved in [775d93c](https://github.com/GreptimeTeam/greptimedb/pull/5626/commits/775d93cbe20fba19299e69dae4321b0c6ea12f3d)